src/librustc/mir/visit.rs

```diff
@@ -786,3 +804,58 @@ pub enum LvalueContext<'tcx> {
     StorageLive,
     StorageDead,
 }
+
+impl<'tcx> LvalueContext<'tcx> {
+    pub fn is_storage_marker(&self) -> bool {
+        match *self {
+            LvalueContext::StorageLive | LvalueContext::StorageDead => true,
+            _ => false,
+        }
+    }
+
+    pub fn is_storage_live_marker(&self) -> bool {
+        match *self {
+            LvalueContext::StorageLive => true,
+            _ => false,
+        }
+    }
+
+    pub fn is_storage_dead_marker(&self) -> bool {
+        match *self {
+            LvalueContext::StorageDead => true,
+            _ => false,
+        }
+    }
+
+    /// Returns true if this lvalue context represents a mutation (def).
+    pub fn is_def(&self) -> bool {
```

Comment:
@pcwalton my feeling is that `def` is the right term, it's just overloaded. Without context, I interpreted it first as "definition" as in `def_id`, versus definition as in "assignment to a variable".